test/jest.browser.setup.ts

```diff
@@ -11,7 +12,20 @@ beforeAll(async () => {
     serverOptions: {
       webpackConfig,
     },
-  })
+  }).then(
+    // NOTE: when upstream patch is fixed, remove this last promise handler
+    (browser) => {
+      patchServerConnectionInfo(
+        // note: we know server connection is up because createBrowser waited `until`
+        // it was up, meaning PreviewServer.listen() set `this.connectionInfo`
+        server as NonNullable<typeof server> & {
+          connectionInfo: NonNullable<typeof server['connectionInfo']>
+        }
+      )
+
+      return browser
+    }
+  )
```

Comment:
Here is where we mutate `server` (see previous comment on `patched/PageWithPreviewServer.ts`) to fix `server.connectionInfo.url`.

Reply:
Do I get this right that we can remove this mutation once your change to `page-with` is published?

Reply:
Yes, that's correct! It's now gone, since it was part of the dropped commit adding the patch. :)